jaeger_client/tracer.py

```diff
@@ -157,10 +157,12 @@ def start_span(self,
                     tags = tags or {}
                     for k, v in six.iteritems(sampler_tags):
                         tags[k] = v
-            elif self.is_debug_allowed(operation_name):  # have debug id
+            elif parent.debug_id and self.is_debug_allowed(operation_name):
```

Comment:
I think this conflates `debug_id` with debug flag. A debug ID is a specific header along the lines of `jaeger-debug-id: foo`, which then associates an ID `"foo"` with a new span. That span also sets its debug flag to true (assuming throttling allows it). In retrospect, maybe we should revert this change.

Reply:
where do you se a disconnect? The only way we get to this part is if we have no parent OR the parent is empty but has a debug_id tag set. Previously those were the only condition, now you can also get here with the parent empty (as in: no trace id), but having the baggage. Yet it may not have the debug_id tag. Previously this `elif` part always implied there are the debug_id tag, now we need to check for it.

Reply:
OK sorry my confusion here was based on a misunderstanding here regarding the necessity of a parent. I had thought you were implying that all debug flags must be propagated from a real span object, not just a header. I forgot the header is transformed into an empty span context with a debug ID, but nothing else.